### DISCUSSION TITLE
Authenticating request for API key

### DISCUSSION THREAD

**Discussion by parkerbroz** [dyc3/steamguard-cli#356] [Q&A] [answered]
To accept trade offers, the account needs an API key. To generate an API key, you need "confirmation via your Steam Guard Mobile Authenticator". steamguard-cli is the only authenticator attached to this account.

If possible, how can I confirm API key generation? Thank you for any help.

**Comment by dyc3** [comment]
If you're referring to the Steam web api key ([the one you get here](https://steamcommunity.com/dev/apikey)), that is **not** necessary for steamguard-cli to work. All the APIs that steamguard-cli uses simply requires logging in to get access tokens (different from the web api key).

If you are trying to accept a confirmation that is not a trade or market confirmation, those should also work with steamguard-cli. You interact with them the same way: run `steamguard trade`.

**Reply by parkerbroz** [reply]
I appreciate the quick response. `steamguard trade` yields ` >[ ] Unknown(9) - Authorize a developer API Key with access to your account`. Am I missing something here?

**Reply by dyc3** [reply]
Yeah. That's how it's supposed to look. At the top of the terminal, you should see the key bindings. You can press <kbd>a</kbd> to mark it as accepted, and then <kbd>enter</kbd> to send the confirmation.

**Reply by dyc3** [reply, marked as answer]
But thanks for reporting this, I didn't know that creating a api key requires a confirmation now. I'll fix it so it doesn't say `Unknown(9)` in the next release.

**Reply by parkerbroz** [reply]
Ah I see it is behaving as intended. The `Unknown(9)` made me think it was an error somehow. Thanks!
